src/cli-sdk/src/commands/cache.ts

```diff
@@ -0,0 +1,306 @@
+  }
+  error(e: unknown) {
+    throw e
+  }
+}
+
+export const views: Views<void | CacheMap> = {
+  human: CacheView,
+}
+
+const usageDef = {
+  command: 'cache',
+  usage: '<command> [flags]',
+  description: 'Work with vlt cache folders',
+
+  subcommands: {
+    add: {
+      usage: '<package-spec> [<package-spec>...]',
+      description:
+        'Resolve the referenced package identifiers and ensure they are cached',
+    },
+
+    ls: {
+      usage: '[<key>...]',
+      description: 'Show cache entries',
+    },
+
+    clean: {
+      usage: '[<key>...]',
+      description: 'Purge expired cache entries',
+    },
+
+    delete: {
+      usage: '<key> [<key>...]',
+      description: 'Purge items explicitly, whether expired or not',
+    },
+
+    'delete-before': {
+      usage: '<date>',
+      description: 'Purge all cache items from before a given date',
+    },
+
+    'delete-all': {
+      usage: '',
+      description:
+        'Delete the entire cache folder to make vlt slower',
+    },
+  },
+} as const satisfies CommandUsageDefinition
+
+export const usage: CommandUsage = () => commandUsage(usageDef)
+
+export const command: CommandFn<void | CacheMap> = async conf => {
+  const [sub, ...args] = conf.positionals
+  switch (sub) {
+    case 'ls':
+      return ls(conf, args, view)
+
+    case 'add':
+      return add(conf, args, view)
+
+    case 'clean':
+      return clean(conf, args, view)
+
+    case 'delete':
+      return deleteKeys(conf, args, view)
+
+    case 'delete-before':
+      return deleteBefore(conf, args, view)
+
+    case 'delete-all':
+      return deleteAll(conf, args, view)
+
+    default: {
+      throw error('Unrecognized cache command', {
+        code: 'EUSAGE',
+        found: sub,
+        validOptions: ['ls', 'add', 'clean'],
+      })
+    }
+  }
+}
+
+const ls = async (
+  conf: LoadedConfig,
+  keys: string[],
+  view?: CacheView,
+): Promise<CacheMap> =>
+  keys.length ?
+    await fetchKeys(
+      conf,
+      keys,
+      (entry: CacheEntry, key: string) => {
+        view?.stdout(
+          key.includes(' ') ? JSON.stringify(key) : key,
+          entry,
+        )
+        return true
+      },
+      view,
+    )
+  : await fetchAll(conf, (_, key) => {
+      //view?.stdout(key)
+      view?.stdout(key.includes(' ') ? JSON.stringify(key) : key)
+      return true
+    })
+
+const fetchAll = async (
+  conf: LoadedConfig,
+  test: (entry: CacheEntry, key: string, val: Buffer) => boolean,
+) => {
+  const rc = new RegistryClient(conf.options)
+  const { cache } = rc
+  const map: CacheMap = {}
+  for await (const [key, val] of cache) {
+    const entry = CacheEntry.decode(val)
+    if (!test(entry, key, val)) continue
+    map[key] = entry.toJSON()
+  }
+  return map
+}
+
+const fetchKeys = async (
+  conf: LoadedConfig,
+  keys: string[],
+  test: (entry: CacheEntry, key: string, buf: Buffer) => boolean,
+  view?: CacheView,
+) => {
+  const rc = new RegistryClient(conf.options)
```

Comment:
is there a `RegistryClient` instance in `conf` that can be used here instead of the new instance? not a necessary change but I wonder if there is a reusable instance already.

Reply:
No, we don't create a RegistryClient, though there is one attached to the packageInfo client that's already there. I could use that 👍